src/edu/csus/ecs/pc2/core/report/ResultsCompareReport.java

```diff
@@ -0,0 +1,369 @@
+
+    private String pc2ResultsDir = null;
+
+    private IInternalContest contest;
+
+    private IInternalController controller;
+
+    private Log log;
+
+    private Filter filter = new Filter();
+
+    private boolean fullDetails = false;
+    
+    private String missingSourceMessage = ComparisonState.MISSING_SOURCE.toString(); 
+    
+    private String missingTargetMessage = ComparisonState.MISSING_TARGET.toString();
+
+    /**
+     * 
+     */
+    private static final long serialVersionUID = -796328654541676730L;
+
+    public ResultsCompareReport(IInternalContest contest, IInternalController controller, String primaryCCSResultsDir, String pc2ResultsDir, boolean fullDetails) {
+        this(contest, controller, primaryCCSResultsDir, pc2ResultsDir, fullDetails, null, null);
+    }
+    
+    public ResultsCompareReport(IInternalContest contest, IInternalController controller, String primaryCCSResultsDir, String pc2ResultsDir, boolean fullDetails, String missingSourceMessage,
+            String missingTargetMessage) {
+        super();
+        this.primaryCCSResultsDir = primaryCCSResultsDir;
+        this.pc2ResultsDir = pc2ResultsDir;
+        this.fullDetails = fullDetails;
+        if (missingTargetMessage != null) {
+            this.missingTargetMessage = missingTargetMessage;
+        }
+        if (missingSourceMessage != null) {
+            this.missingSourceMessage = missingSourceMessage;
+        }
+        setContestAndController(contest, controller);
+    }
+    
+
+    public ResultsCompareReport() {
+        ;
+    }
+
+    @Override
+    public void setContestAndController(IInternalContest inContest, IInternalController inController) {
+        this.contest = inContest;
+        this.controller = inController;
+        log = controller.getLog();
+
+        /**
+         * Fetch directories from ClientSettings.
+         */
+        ClientSettings settings = contest.getClientSettings(contest.getClientId());
+
+        if (pc2ResultsDir == null) {
+            String pc2rsdir = settings.getProperty(ClientSettings.PC2_RESULTS_DIR);
+            pc2ResultsDir = pc2rsdir;
+        }
+
+        if (primaryCCSResultsDir == null) {
+            String primaryResDir = settings.getProperty(ClientSettings.PRIMARY_CCS_RESULTS_DIR);
+            primaryCCSResultsDir = primaryResDir;
+        }
+
+    }
+
+    @Override
+    public String getPluginTitle() {
+        return "Results Compare Report";
+    }
+
+    @Override
+    public void createReportFile(String filename, Filter inFilter) throws IOException {
+        PrintWriter printWriter = new PrintWriter(new FileOutputStream(filename, false), true);
+        filter = inFilter;
+
+        try {
+            printHeader(printWriter);
+
+            try {
+                writeReport(printWriter);
+            } catch (Exception e) {
+                printWriter.println("Exception in report: " + e.getMessage());
+                e.printStackTrace(printWriter);
```

Comment:
Since `setContestAndController()` has established a `log`, this exception condition should be logged.